tests/config/test_config.yml

```diff
@@ -7,399 +7,430 @@
-
-  - name: set maxusername to 33
-    ipaconfig:
-      ipaadmin_password: SomeADMINpassword
-      maxusername: 33
-    register: result
-    failed_when: not result.changed or result.failed
-
-  - name: set maxusername to 33, again.
-    ipaconfig:
-      ipaadmin_password: SomeADMINpassword
-      maxusername: 33
-    register: result
-    failed_when: result.changed or result.failed
-
-  - name: set maxhostname to 77
-    block:
-      - ipaconfig:
-          ipaadmin_password: SomeADMINpassword
-          maxhostname: 77
+  - block:
+      # Retrieve current configuration.
+      - name: return current values of the global configuration options
+        ipaconfig:
+          ipaadmin_password: SomeADMINpassword
+        register: previousconfig
+
+      - debug:
+          var: previousconfig
+
+      # setup environment.
+      - name: create test group
+        ipagroup:
+          ipaadmin_password: SomeADMINpassword
+          name: somedefaultgroup
+
+      - name: Ensure the default e-mail domain is ipa.test.
+        ipaconfig:
+          ipaadmin_password: SomeADMINpassword
+          emaildomain: ipa.test
+
+      - name: set default shell to '/bin/sh'
+        ipaconfig:
+          ipaadmin_password: SomeADMINpassword
+          defaultshell: /bin/sh
+
+      - name: set default group
+        ipaconfig:
+          ipaadmin_password: SomeADMINpassword
+          defaultgroup: ipausers
+
+      - name: set default home directory
+        ipaconfig:
+          ipaadmin_password: SomeADMINpassword
+          homedirectory: /home
+
+      - name: clear pac-type
+        ipaconfig:
+          ipaadmin_password: SomeADMINpassword
+          pac_type: ""
+
+      - name: set maxhostname to 255
+        block:
+          - ipaconfig:
+              ipaadmin_password: SomeADMINpassword
+              maxhostname: 255
+        when: ipa_version is version('4.8.0', '>=')
+
+      - name: set maxusername to 45
+        ipaconfig:
+          ipaadmin_password: SomeADMINpassword
+          maxusername: 45
+
+      - name: set pwdexpnotify to 0
+        ipaconfig:
+          ipaadmin_password: SomeADMINpassword
+          pwdexpnotify: 0
+
+      - name: set searchrecordslimit to 10
+        ipaconfig:
+          ipaadmin_password: SomeADMINpassword
+          searchrecordslimit: 10
+
+      - name: set searchtimelimit to 1
+        ipaconfig:
+          ipaadmin_password: SomeADMINpassword
+          searchtimelimit: 1
+
+      - name: clear configstring
+        ipaconfig:
+          ipaadmin_password: SomeADMINpassword
+          configstring: ""
+
+      - name: set configstring to AllowNThash
+        ipaconfig:
+          ipaadmin_password: SomeADMINpassword
+          configstring: 'KDC:Disable Lockout'
+
+      - name: set selinuxusermapdefault
+        ipaconfig:
+          ipaadmin_password: SomeADMINpassword
+          selinuxusermapdefault: "staff_u:s0-s0:c0.c1023"
+        check_mode: yes
```

Comment:
Shouln't check_mode be tested additionally to a change that was really done?

Reply:
This is here by mistake. I'll fix it.

Reply:
fixed.